### ISSUE TITLE
Long Fine-Tuning Time

### ISSUE THREAD

**Issue by HengguangZhou** [PKU-YuanGroup/LLaVA-CoT#19]
Hi, thank you for the excellent work and the repository! I am currently trying to replicate the fine-tuning results using 4 A6000 GPUs. I have read https://github.com/PKU-YuanGroup/LLaVA-CoT/issues/8 and was trying to follow the instruction on https://github.com/meta-llama/llama-recipes/blob/main/recipes/quickstart/finetuning/finetune_vision_model.md, along with the hyperparameter you provided in the paper. After following the setup instructions, I found the training time seems significantly longer than expected.

I noticed you mentioned in that the total training time is ~15 hours https://github.com/PKU-YuanGroup/LLaVA-CoT/issues/7. However, in my case it's taking around 40 hours for just one epoch. So I was wondering:
1. Is this training time expected for my configuration, or might there be an issue with my setup?
2. Are there any techniques or optimizations you recommend to improve the training speed?
3. Do you have plans to release the fine-tuning script?

Thanks in advance!

**Comment by XuGW-Kevin** [comment]
Hi, thanks for raising the issue! We used 8 * H100 GPUs in our experiments, and perhaps this is the main reason for speed discrepancy.  The fine-tuning script we used is the same as that provided in  https://github.com/meta-llama/llama-recipes/blob/main/recipes/quickstart/finetuning/finetune_vision_model.md.

**Comment by XuGW-Kevin** [comment]
torchrun --nnodes 1 --nproc_per_node 4  recipes/quickstart/finetuning/finetuning.py --enable_fsdp --lr 1e-5  --num_epochs 3 --batch_size_training 4 --model_name meta-llama/Llama-3.2-11B-Vision-Instruct --dist_checkpoint_root_folder ./finetuned_model --dist_checkpoint_folder fine-tuned  --use_fast_kernels --dataset "custom_dataset" --custom_dataset.test_split "test" --custom_dataset.file "recipes/quickstart/finetuning/datasets/the_dataset.py"  --run_validation False --batching_strategy padding

**Comment by HengguangZhou** [comment]
I believe I was using the same settings, and as you mentioned, the GPU configuration seems to be the primary factor. Thank you for the information!
